streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -227,6 +232,41 @@ public void to(Serde<K> keySerde, Serde<V> valSerde, String topic) {
         topology.addSink(name, topic, keySerializer, valSerializer, streamPartitioner, this.name);
     }
 
+
+    @Override
+    public KStream<K, V> print() {
+        return print(null, null);
+    }
+
+    @Override
+    public KStream<K, V> print(Serde<?> keySerde, Serde<?> valSerde) {
+        String name = topology.newName(PRINTING_NAME);
+        topology.addProcessor(name, new KStreamPrinter<>(keySerde, valSerde), this.name);
+        return new KStreamImpl<>(topology, name, sourceNodes);
+    }
+
+
+    @Override
+    public KStream<K, V> writeAsText(String filePath) {
+        return writeAsText(filePath, null, null);
+    }
+
+    @Override
+    public KStream<K, V> writeAsText(String filePath, Serde<?> keySerde, Serde<?> valSerde) {
+        String name = topology.newName(PRINTING_NAME);
+
+        try {
+
+            PrintStream printStream = new PrintStream(new FileOutputStream(filePath));
+            topology.addProcessor(name, new KStreamPrinter<>(printStream, keySerde, valSerde), this.name);
+            return new KStreamImpl<>(topology, name, sourceNodes);
+
+        } catch (FileNotFoundException e) {
+            String message = "Unable to write stream to file at [" + filePath + "]";
+            throw new IllegalStateException(message, e);
```

Comment:
Maybe return TopologyBuilderException since it will not be thrown in "runtime" after `KafkaStreams.start()` is triggered, but before that when the topology was built.